### PR TITLE
Change SpanProcessor to use Context.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Caveats:
- Lose some source format in logs;
- Some recorded metrics may not include the source, but I left TODOs to fix that.

### CODE REVIEW

File: internal/collector/processor/nodebatcher/node_batcher.go [hunk 1]
```diff
@@ -172,7 +171,6 @@ type nodeBatcher struct {
 	currBatch     *batch
 	node          *commonpb.Node
 	resource      *resourcepb.Resource // TODO(skaris) remove when resource is added to span
-	spanFormat    string
```
Comment (songy23) [inline internal/collector/processor/nodebatcher/node_batcher.go]: Why remove this?

File: internal/collector/processor/tailsampling/processor.go [hunk 2]
```diff
@@ -189,7 +192,7 @@ func (tsp *tailSamplingSpanProcessor) ProcessSpans(td data.TraceData, spanFormat
 	idToSpans := make(map[traceKey][]*tracepb.Span)
 	for _, span := range td.Spans {
 		if len(span.TraceId) != 16 {
-			tsp.logger.Warn("Span without valid TraceId", zap.String("spanFormat", spanFormat))
+			tsp.logger.Warn("Span without valid TraceId")
```
Comment (pjanotti) [inline internal/collector/processor/tailsampling/processor.go]: That information should be on the ctx, it seems good information to be on the warning message.

File: internal/collector/processor/nodebatcher/node_batcher.go (hunk 1 above)
Comment (pjanotti) [reply internal/collector/processor/nodebatcher/node_batcher.go]: Why not a reference to context in the place of the field?

**Comment by bogdandrutu** [comment]
@songy23 @pjanotti I think I will re-do this PR with a different change because:
1. No way to access the source name from ctx to log (we actually work on the ability to add tags to the log messages but not available yet).
1. Avoid changing the collector stats for the moment.

Proposed alternative:
1. Add source format to the TraceData for the moment, with a comment that this will be removed in the near feature after the unification on metrics/stats.
2. Change the SpanProcessor to use (Context, TraceData) and don't change any logic around format name because this will still be available.

File: internal/collector/processor/nodebatcher/node_batcher.go [hunk 3]
```diff
@@ -421,6 +415,7 @@ func (b *batch) add(spans []*tracepb.Span) (cut bool, closed bool) {
 		b.grow(openTill)
 	}
 
+	// Why not: copy(b.items.Load().([]*tracepb.Span)[openFrom:openTill], spans)?
```
Comment (sjkaris) [inline internal/collector/processor/nodebatcher/node_batcher.go]: @bogdandrutu we should just supplant this all this logic with something simpler as you said. https://github.com/census-instrumentation/opencensus-service/pull/487 replaces this, addressing the PR comments there now

Comment (bogdandrutu) [reply internal/collector/processor/nodebatcher/node_batcher.go]: As I mentioned in my previous comment I feel like I am changing too much in this PR and I want to go with that proposal of still passing the source name for the moment in the TraceData then remove it later from there.

**Comment by bogdandrutu** [comment]
I would like to get an agreement on what you think about the latest proposal of not removing the source name for the moment and do that later, as I mentioned in the comment https://github.com/census-instrumentation/opencensus-service/pull/485#issuecomment-471740800

**Comment by songy23** [comment]
> Add source format to the TraceData for the moment, with a comment that this will be removed in the near feature after the unification on metrics/stats.
Change the SpanProcessor to use (Context, TraceData) and don't change any logic around format name because this will still be available.

+1 to this. For now let's leave the span format name.

**Comment by pjanotti** [comment]
I agree with the proposal, adding to td seems a reasonable compromise.

**Comment by bogdandrutu** [comment]
Closing this in favor of https://github.com/census-instrumentation/opencensus-service/pull/489

**Pull request closed without merging**